examples/compose/plot_compare_reduction.py

```diff
@@ -92,7 +92,8 @@
 
 # %%
 # Caching transformers within a ``Pipeline``
-###############################################################################
+# ------------------------------------------
```

Comment:
Why change this section and not "Illustration of ``Pipeline`` and ``GridSearchCV``"? Maybe this is an oversight?

Reply:
More generally, why change the header underline style? This seems unrelated to the purpose of this PR. I am not opposed to it but I wonder what are guidelines to choose the symbol used for underlines in examples section headers.

Reply:
Ah I didn't realize that `####` also creates valid header, so the problem here is that there must be a space after the first `#`. I think I will stick to the original, thanks for noticing :)